find_max.rb

```diff
@@ -0,0 +1,19 @@
+
+# Find the maximum element between index 0 and max_index, exclusive
+
+def find_max(array, max_index)
+  max = nil
+  if max_index == 0
```

Comment:
If we check for max_index > 0 here and then execute the code in the else statement, _max_ can be declared and returned from within the same statement.